db/cache_mutation_reader.hh

```diff
@@ -198,13 +198,13 @@ class cache_mutation_reader final : public mutation_reader::impl {
     }
 
     gc_clock::time_point get_read_time() {
```

Comment:
> db/row_cache: migrate to tombstone_gc_before_getter

> tombstone_gc_before_getter has value semantics, so tombstone_gc_state*
> is replaced by std::optional<tombstone_gc_before_getter>.

Isn't something like no_gc equivalent? e.g tombstone_gc_before_getter is already self-optional?